java/fury-core/src/test/java/org/apache/fury/serializer/PrimitiveSerializersTest.java

```diff
@@ -54,4 +59,62 @@ public void testUint16Serializer() {
     assertThrows(IllegalArgumentException.class, () -> serializer.xwrite(buffer, -1));
     assertThrows(IllegalArgumentException.class, () -> serializer.xwrite(buffer, 65536));
   }
+
+  @Data
+  @AllArgsConstructor
+  public static class PrimitiveStruct {
+    byte byte1;
+    byte byte2;
+    char char1;
+    char char2;
+    short short1;
+    short short2;
+    int int1;
+    int int2;
+    long long1;
+    long long2;
+    float float1;
+    float float2;
+    double double1;
+    double double2;
+  }
+
+  @Test(dataProvider = "compressNumberAndCodeGen")
+  public void testPrimitiveStruct(boolean compressNumber, boolean codegen) {
+    PrimitiveStruct struct =
+        new PrimitiveStruct(
+            Byte.MIN_VALUE,
+            Byte.MIN_VALUE,
+            Character.MIN_VALUE,
+            Character.MIN_VALUE,
+            Short.MIN_VALUE,
+            Short.MIN_VALUE,
+            Integer.MIN_VALUE,
+            Integer.MIN_VALUE,
+            Long.MIN_VALUE,
+            Long.MIN_VALUE,
+            Float.MIN_VALUE,
+            Float.MIN_VALUE,
+            Double.MIN_VALUE,
+            Double.MIN_VALUE);
+    if (compressNumber) {
+      FuryBuilder builder =
```

Comment:
Nit: We can move the common part of FuryBuilder before the if branch, so that if and else can share this part